test/e2e/storage/testsuites/provisioning.go

```diff
@@ -268,6 +297,15 @@ func TestDynamicProvisioning(t StorageClassTest, client clientset.Interface, cla
 		Expect(err).NotTo(HaveOccurred())
 	}
 
+	By(fmt.Sprintf("xxxxxxx dataSource %v", claim.Spec.DataSource))
+	if claim.Spec.DataSource != nil {
+		framework.Logf("check datasource %q/%q", claim.Namespace, claim.Name)
+
+		By("checking the created volume whether has the pre-populated data")
+		command := fmt.Sprintf("grep '%s' /mnt/test/initialData", claim.Namespace)
+		runInPodWithVolume(client, claim.Namespace, claim.Name, t.NodeName, command, t.NodeSelector, t.ExpectUnschedulable)
+	}
+
```

Comment:
Please put this check into a local function of your "should provision storage with snapshot data source" and then assign that function to `input.testCase.PvCheck`. That way it is obvious that the check belongs to that test case, and only that testcase.

Reply:
now the PvCheck only one argument `v1.PersistentVolume`, this check need also `v1.PersistentVolumeClaim` argument. So we need  refactor the `PvCheck` func first, and I have saw your do this work in PR https://github.com/kubernetes/kubernetes/pull/72002, can we merge this first and fix it after your pr get merged.

Reply:
Okay.